### ISSUE TITLE
Cannot auth if vscode.dev is in iframe

### ISSUE THREAD

**Issue by DawidMazur** [microsoft/vscode#150152]
ADD ISSUE DESCRIPTION HERE

Hi
I noticed that there is a problem with github authentication when the page is in iframe, the external window is displayed for a while, it disappears but authentication doesn't go any further.

Tested on localhost (by xampp with standard configuration)

Version: 1.67.2
Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36
Embedder: vscode.dev

**Comment by TylerLeonhardt** [comment]
Yeah embedding vscode.dev in an iframe is not a supported scenario for us at this time.

We may explore this in the future but it would take a considerable amount of work to get right.

cc @bamurtaugh do we have a issue around this? Otherwise, I can keep this open.

**Comment by bamurtaugh** [comment]
@TylerLeonhardt Let's keep this issue open - I don't think we have another.

**Comment by claranceliberi** [comment]
Nuxt team has been working on nuxt-dev-tools project, we are trying to embed vscode.dev in our devtools  https://github.com/nuxt/devtools/issues/38 so that users can edit codes directly in the browser and get all the vs code benefits while debugging or developing. 

At the moment, this can't be achieved with vscode.dev due to this issue. we would like to know the way forwad

**Comment by Sandros94** [comment]
> Nuxt team has been working on nuxt-dev-tools project, we are trying to embed vscode.dev in our devtools [nuxt/devtools#38](https://github.com/nuxt/devtools/issues/38) so that users can edit codes directly in the browser and get all the vs code benefits while debugging or developing.
> 
> At the moment, this can't be achieved with vscode.dev due to this issue. we would like to know the way forwad

Just to provide two simple use cases for this:
- Developer on a single-screen machine (like a laptop): they have both vscode and browser open and usually they need to alt-tab back and forward to check things. Embedding vscode inside an iframe of `nuxt-devtools` we could have vscode floating in our browser window and speed up the development.
- Developer on the go: the developer only has a windows desktop and a mobile tablet, since there is no direct support for `code-server` on windows, vscode tunnel is the only other option. They could open a tunnel connection for both vscode and node port, and opening up only the second link, while being able to use vscode inside the iframe in `nuxt-devtools`. Having only one browser tab for this job would greatly improve the battery consumption on mobile devices and let them develop longer.
